packages/adapters-library/src/core/apy-calculators/BalanceOfApyCalculator.ts

```diff
@@ -0,0 +1,122 @@
+import { AVERAGE_BLOCKS_PER_DAY } from '../constants/AVERAGE_BLOCKS_PER_DAY'
+import { ApyCalculation, ApyCalculator, EvmApyArgs } from './ApyCalculator'
+import { computeApr, computeApy } from './common'
+
+/**
+ * Calculates the APY of a position by calculating the fees as the difference of balance in
+ * the underlying token between a given day and the day before.
+ *
+ * Is compatible with ANY DeFi protocol where the position is tracked by a single amount in wei (for instance the balanceOf an ERC20),
+ * and where the balance varies only due to accruing fees/interest ( assuming no deposit or withdrawal).
+ *
+ * We can translate this mathematically to:
+ * balance(t') = balance(t) + fees(t)
+ *
+ * Compatible protocols / products are:
+ * - Stablecoins
+ * - Lending (aToken, cToken, ...)
+ * - LSTs
+ */
+export class BalanceOfApyCalculator implements ApyCalculator<EvmApyArgs> {
+  // Duration in days of the period where we look at the fees earned. The APY and APR are then annualized based on this period.
+  public static readonly PERIOD_DURATION_DAYS = 1
+
+  // How many periods there is in a year
+  public static readonly FREQUENCY = 365
+
+  public async getApy({
+    userAddress,
+    blockNumber,
+    protocolTokenAddress,
+    chainId,
+    adapter,
+  }: EvmApyArgs): Promise<ApyCalculation> {
```

Comment:
I noticed you are passing the adapter here.

Incase you didnt know the AdapterController has a method where you can pass just the protocol token address and chainId and itll return the matching adapter.

```
  async fetchTokenAdapter(
    chainId: Chain,
    tokenAddress: string,
  ): Promise<IProtocolAdapter | undefined> {
    // Deferred promise so that only the first execution path does the work
    if (!this.protocolTokens) {
      this.protocolTokens = this.buildProtocolTokens()
    }

    const protocolTokens = await this.protocolTokens

    return protocolTokens.get(chainId)?.get(tokenAddress)
  }

```